bridges/snowbridge/pallets/system-frontend/src/lib.rs

```diff
@@ -105,49 +119,78 @@ pub mod pallet {
 		/// Convert versioned location failure
 		UnsupportedLocationVersion,
 		/// Check location failure, should start from the dispatch origin as owner
-		OwnerCheck,
+		InvalidAssetOwner,
 		/// Send xcm message failure
 		Send,
 		/// Withdraw fee asset failure
 		FundsUnavailable,
+		/// Convert to reanchored location failure
+		LocationConversionFailed,
 	}
 
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
 		/// Call `create_agent` to instantiate a new agent contract representing `origin`.
-		/// - `origin`: Can be any sovereign `Location`
+		/// - `origin`: `Location` can be from either of:
+		/// 	a. any kind of location from a sibling parachain
+		/// 	b. sudo from local, location will be `Here` in this case.
+		/// 	c. a signed origin from local, location will be `(0,[AccountId32()])` in this case.
 		/// - `fee`: Fee in Ether paying for the execution cost on Ethreum
 		#[pallet::call_index(1)]
 		#[pallet::weight(T::WeightInfo::create_agent())]
 		pub fn create_agent(origin: OriginFor<T>, fee: u128) -> DispatchResult {
 			let origin_location = T::CreateAgentOrigin::ensure_origin(origin)?;
 
-			Self::burn_fees(origin_location.clone(), fee)?;
+			// Ignore fee charges for sudo call
+			if !origin_location.eq(&Here.into()) {
+				// Burn Ether Fee for the cost on ethereum
+				Self::burn_for_teleport(&origin_location, &(T::FeeAsset::get(), fee).into())?;
+
+				// Burn RemoteExecutionFee for the cost on bridge hub
+				Self::burn_for_teleport(&origin_location, &T::RemoteExecutionFee::get())?;
+			}
```

Comment:
This just won't work (see problems below). Its really half baked. Please forget about `sudo` support for now, and get the permissionless version working and tested 100%.

1. You're teleporting the remote execution fee, and so it always needs to be burned locally.
2. Its not as simple as checking `location == Here` for governance. You need to use a `EnsureOrigin` checker configured by the Runtime.
3. We may want some snowbridge treasury account to fund fees for root, rather than skipping fees.

Reply:
All make sense!  Remove `sudo` in https://github.com/vgeddes/polkadot-sdk/pull/6/commits/e7c056ac98152a9956a07d51937e2eccb20e57b3

Reply:
https://github.com/vgeddes/polkadot-sdk/pull/6/commits/55bf251a5b2635a0e80a8377d3bdf8e11b76581c